pages/coaching-dashboard.tsx

```diff
@@ -20,26 +27,28 @@ const coachingDashboard = () => {
     <div className="flex flex-col p-4 m-4 ">
       <p className="mb-8 text-3xl">Coaching Dashboard</p>
       <h2 className="mb-4">Enrolled Students</h2>
+
       <div className="grid grid-cols-1 gap-4 sm:grid-cols-3">
-        {data.users.map((it) => (
-          <Link href={"profile/" + it.link}>
-            <div className="">
-              <ProfileDetailCard
-                avatar={
-                  it.profile_image == null
-                    ? "../../images/logo-2.png"
-                    : it.profile_image
-                }
-                name={it.name}
-                joinDate={it.created_at}
-                badges={it.badges_earned}
-                currentFocus={""}
-                nextGoal={""}
-                link={it.link}
-              />
-            </div>
-          </Link>
-        ))}
+        {data.users.length > 0 &&
```

Comment:
let's map from state instead of the query response data

right now you aren't using your redux state at all